transport/tchannel/header.go

```diff
@@ -22,14 +22,51 @@ package tchannel
 
 import (
 	"encoding/binary"
+	"fmt"
 	"io"
+	"strings"
 
+	"golang.org/x/net/context"
+
+	"github.com/yarpc/yarpc-go/internal/baggage"
 	"github.com/yarpc/yarpc-go/transport"
 	"github.com/yarpc/yarpc-go/transport/tchannel/internal"
 
 	"github.com/uber/tchannel-go"
 )
 
+// pullBaggage pulls the context headers from the given transport.Headers,
+// deleting them from the original headers map.
+func pullBaggage(headers transport.Headers) transport.Headers {
+	ctxheaders := make(transport.Headers)
+	prefix := strings.ToLower(BaggageHeaderPrefix)
+	prefixLen := len(prefix)
+	for k, v := range headers {
+		if strings.HasPrefix(k, prefix) {
+			key := k[prefixLen:]
+			ctxheaders.Set(key, v)
+			headers.Del(k)
+		}
+	}
+	return ctxheaders
+}
+
+// readRequestHeaders reads headers and baggage from an incoming request.
+func readRequestHeaders(
+	ctx context.Context,
+	format tchannel.Format,
+	getReader func() (tchannel.ArgReader, error),
+) (context.Context, transport.Headers, error) {
+	headers, err := readHeaders(format, getReader)
+	if err != nil {
+		return ctx, nil, err
+	}
+	if ctxheaders := pullBaggage(headers); ctxheaders != nil {
```

Comment:
nit: here and above, can we call it `ctxHeaders`?

also, instead of doing a `nil` check, can we do `if len(ctxHeaders) > 0`